spec/models/git_repository_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'spec_helper'
+
+describe GitRepository do
+  it "no url" do
+    expect { FactoryGirl.create(:git_repository, :dirname => "abc") }.to raise_error(ActiveRecord::RecordInvalid)
+  end
+
+  it "invalid url" do
+    expect { FactoryGirl.create(:git_repository, :url => "abc") }.to raise_error(ActiveRecord::RecordInvalid)
+  end
+
+  it "default dirname" do
+    repo = FactoryGirl.create(:git_repository,
+                              :url => "http://www.something.com/repos/manageiq")
+    expect(repo.dirname).to eq(File.join(MiqAeDatastore::GIT_REPO_DIRECTORY, 'repos/manageiq'))
+  end
+
+  it "passed in dirname" do
+    repo = FactoryGirl.create(:git_repository,
+                              :dirname => '/tmp/repodir',
+                              :url     => "http://www.a.com/repos/manageiq")
+    expect(repo.dirname).to eq('/tmp/repodir')
+  end
+
+  context "repo" do
+    let(:gwt) { instance_double('GitWorktree') }
+    let(:branch_list) { %w(b1 b2) }
+    let(:tag_list) { %w(t1 t2) }
+    let(:info) { {:time => Time.now.utc, :message => "R2D2", :commit_sha => "abcdef"} }
+    let(:branch_info_hash) do
+      {'b1' => {:time => Time.now.utc, :message => "B1", :commit_sha => "abcdef"},
+       'b2' => {:time => Time.now.utc - 5, :message => "B2", :commit_sha => "123456"}
+      }
+    end
+    let(:tag_info_hash) do
+      {'t1' => {:time => Time.now.utc, :message => "T1", :commit_sha => "abc12f"},
+       't2' => {:time => Time.now.utc + 5, :message => "T2", :commit_sha => "123456"}
+      }
+    end
+
+    it "init" do
```

Comment:
As mentioned in the other PR, the tests should be named according to the method under test.  In this case, the method under test seems to be "#refresh" and the case is "when the repo doesn't exist"